packages/apollo-gateway/src/index.ts

```diff
@@ -108,25 +170,91 @@ export class ApolloGateway implements GraphQLService {
     }
 
     this.initializeQueryPlanStore();
+
+    // this will be overwritten if the config provides experimental_updateServiceDefinitions
+    this.updateServiceDefinitions = this.loadServiceDefinitions;
+
+    if (config) {
+      this.updateServiceDefinitions =
+        config.experimental_updateServiceDefinitions ||
+        this.updateServiceDefinitions;
+      // set up experimental observability callbacks
+      this.experimental_didResolveQueryPlan =
+        config.experimental_didResolveQueryPlan;
+      this.experimental_didFailComposition =
+        config.experimental_didFailComposition;
+      this.experimental_didUpdateComposition =
+        config.experimental_didUpdateComposition;
+      this.experimental_pollInterval = config.experimental_pollInterval;
+
+      // ie if they have a pollinterval and a custom loader or a serviceList
+      if (config.experimental_pollInterval) {
+        if (
+          config.experimental_updateServiceDefinitions ||
```

Comment:
why is this necessarily bad? Seems like this would throw a warning even if you're creating a wrapper that looks at the managed config and then has a fallback on error... or do we use a different poll interval for that? 😕

Reply:
This is a reasonable consideration. I think this is safe to remove, since there are plenty of other DIY approaches that aren't dangerous either.

Reply:
Resolved in 76beee89d02a45d186ee2ca60e0f1fead636c6bf